scriptmodules/supplementary/skyscraper.sh

```diff
@@ -0,0 +1,687 @@
+    local local_config
+    local_config=$(readlink -qn "$home/.skyscraper")
+
+    # Handle the cases where the user has an existing Skyscraper installation.
+    if [[ -d "$home/.skyscraper" && "$local_config" != "$configdir/all/skyscraper" ]]; then
+        # We have an existing Skyscraper installation, but not handled by this scriptmodule.
+        # Since the $HOME/.skyscraper folder will be moved, make sure the Local DB and Import folders are moved separately
+        local f_size
+        f_size=$(du --total -sm "$home/.skyscraper/dbs" "$home/.skyscraper/import" 2>/dev/null | grep total | cut -f 1 )
+        printMsgs "console" "INFO: Moving the Local DB and Import folders to new configuration folder (total: $f_size Mb)"
+
+        local folder
+        for folder in dbs import; do
+            mv "$home/.skyscraper/$folder" "$home/.skyscraper-$folder" && \
+                printMsgs "console" "INFO: Moved "$home/.skyscraper/$folder" to "$home/.skyscraper-$folder""
+        done
+
+        # When having an existing installation, changes are the gamelist is in the ROMs folder
+        # Create a GUI config file with this setting pre-set.
+        iniConfig " = " '"' "$configdir/all/skyscraper.cfg"
+        iniSet "use_rom_folder" 1
+    fi
+
+    moveConfigDir "$home/.skyscraper" "$configdir/all/skyscraper"
+
+    # When migrating an existing installation, make sure we move the Local DB and Import folder back the new conf folder
+    for folder in dbs import; do
+        if [[ -d "$home/.skyscraper-$folder" ]]; then
+            printMsgs "console" "INFO: Moving "$home/.skyscraper-$folder" back to configuration folder"
+            mv  "$home/.skyscraper-$folder" "$configdir/all/skyscraper/$folder"
+        fi
+    done
+
+    _init_config_skyscraper
+    chown -R $user:$user "$configdir/all/skyscraper"
+}
+
+function _init_config_skyscraper() {
+    local md_conf_dir="$configdir/all/skyscraper"
+
+    cat > "$md_conf_dir/config.ini.rp-dist" << _EOF_
+[main]
+frontend="emulationstation"
+dbFolder="$md_conf_dir/dbs"
+importFolder="$md_conf_dir/import"
+inputFolder="$romdir"
+unattend="true"
+unattendSkip="true"
+noHints="false"
+threads="2"
+maxFails="30"
+lang="en"
+region="wor"
+videos="false"
+marquees="true"
+wheels="false"
+covers="true"
+screenshots="true"
+artworkXml="$md_conf_dir/artwork.xml"
+
+_EOF_
+
+    # Make sure the `artwork.xml` and other conf file(s) are present, but don't overwrite them on upgrades.
+    # For now we just have one file, but more might come up in the future.
+    local f_conf
+    for f_conf in artwork.xml; do
+        if [[ -f "$md_conf_dir/$f_conf" ]]; then
+            cp "$md_build/$f_conf" "$md_conf_dir/$f_conf.rp-dist"
+        else
+            cp "$md_build/$f_conf" "$md_conf_dir"
+        fi
+    done
+
+    # If we don't have a previous config.ini file, copy the dist one
+    [[ ! -f "$md_conf_dir/config.ini" ]] && cp "$md_conf_dir/config.ini.rp-dist" "$md_conf_dir/config.ini"
+
```

Comment:
Might it be worth using copyDefaultConfig. It could be that this isn't exactly what you need, but with current retropie modules we only create a rp-dist if the standard config is changes from the default. This may be useful or not to you.

Reply:
The idea here was to generate a basic default for a new installation and to save it in case editing it will bungle the config. By default skyscraper has a `config.ini.example` but it can run without.  I can change the suffix to keep the convention for `.rp-dist`.

Reply:
Happy to leave the decision to you. Just wanted to mention the function. Whatever you think is best in this case. Thanks